web/src/app/app.component.ts

```diff
@@ -0,0 +1,160 @@
+declare let require: any;
+declare var web3: any;
+import { Component } from '@angular/core';
+const etherlime = require('etherlime');
+const ethers = require('ethers');
+const ToDo = require('../../../build/ToDoManager.json');
+
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.css']
+})
+
+
+export class AppComponent {
+  title = 'My Decentralized App';
+
+  public successMessage: string;
+  public infoMessage: string;
+  public errorMessage: string;
+  public contractInstance: any;
+  public contractAddress = '0xc9707E1e496C12f1Fa83AFbbA8735DA697cdBf64';
+  public toDoStatus: any;
+  public inProgressStatus: any;
+  public doneStatus: any;
+  public inactiveButton;
+
+  constructor() {
+    this.toDoStatus = [];
+    this.inProgressStatus = [];
+    this.doneStatus = [];
+
+    etherlime.ContractAt(ToDo, this.contractAddress).then((result) => {
+      this.contractInstance = result;
+      this.successMessage = 'The contract has been set and is ready to interact with it!';
+      this.getToDoStatuses();
+    }).catch((e) => {
+      this.errorMessage = e.message;
+    });
+
+  }
+
+  private async getToDoIndex(_toDo) {
+    const indexCounter = await this.contractInstance.indexCounter();
+    for (let i = 0; i < indexCounter.toNumber(); i++) {
+      let toDo = await this.contractInstance.getToDoByIndex(i);
+      if (toDo === _toDo) {
+        return i
+      }
+    }
+  }
+
+  private async getSignerInstance() {
+    const provider = new ethers.providers.Web3Provider(web3.currentProvider);
+    const signer = await provider.getSigner();
+    const contract = new ethers.Contract(this.contractAddress, ToDo.abi, provider)
+    const instance = await contract.connect(signer);
+    return instance;
+  }
+
+  public async addToDo(todo) {
+    try {
+      this.inactiveButton = true;
+      await this.contractInstance.addToDo(todo.value);
+      this.addInfoMessage('ToDo has been added!');
+      this.toDoStatus.push(todo.value);
+      this.inactiveButton = false;
+    } catch (e) {
+      this.addInfoMessage(`Transaction failed. Are you sure you hadn't already added this ToDo! ${e.message}`);
+      this.inactiveButton = false;
+    }
+  }
+
+  public async assignToDo(assignedTodo) {
+    try {
+      const instance = await this.getSignerInstance();
+      await instance.assignToDo(assignedTodo);
+      this.addInfoMessage('ToDo was assigned to you!')
+    } catch (e) {
+      this.addInfoMessage(`${e.message}. You can not assign to a ToDo if it is not added the list!`)
+    }
+  }
+
+  public async changeToDoStatus(todo, destination) {
```

Comment:
I'd like to see methods like `moveToInProgress` and `moveToDone` that utilize this method instead of hardcoding some values("progress", "done") in the html